### PR TITLE
Feature/reverse collapse rows

### PR DESCRIPTION
**Pull request by gregpriday**
Resolves GH-129. Also makes RTL support more efficient by changing where the order is reversed.

### CODE REVIEW

File: siteorigin-panels.php [hunk 1]
```diff
@@ -732,11 +732,14 @@ function siteorigin_panels_generate_css($post_id, $panels_data = false){
 			));
 		}
 
+
+		$collapse_order = !empty( $grid['style']['collapse_order'] ) ? $grid['style']['collapse_order'] : ( !is_rtl() ? 'left-top' : 'right-top' );
```
Comment (braamgenis) [inline siteorigin-panels.php]: Shouldn't `is_rtl()` still make it switch even if the collapse order was set by the user?

File: siteorigin-panels.php [hunk 2]
```diff
@@ -925,7 +928,8 @@ function siteorigin_panels_render( $post_id = false, $enqueue_css = true, $panel
 		}
 	}
 
-	if( is_rtl() ) $panels_data = siteorigin_panels_make_rtl( $panels_data );
+	// A filter to change the panels_data right before the content is rendered.
+	$panels_data = apply_filters( 'siteorigin_panels_data_before_render', $panels_data, $post_id );
```
Comment (braamgenis) [inline siteorigin-panels.php]: Shouldn't this filter now be added somewhere to do whatever `siteorigin_panels_make_rtl` was doing?

File: siteorigin-panels.php [hunk 3]
```diff
@@ -1008,6 +1012,12 @@ function siteorigin_panels_render( $post_id = false, $enqueue_css = true, $panel
 		$row_style_wrapper = siteorigin_panels_start_style_wrapper( 'row', $style_attributes, !empty($panels_data['grids'][$gi]['style']) ? $panels_data['grids'][$gi]['style'] : array() );
 		if( !empty($row_style_wrapper) ) echo $row_style_wrapper;
 
+		$collapse_order = !empty( $panels_data['grids'][$gi]['style']['collapse_order'] ) ? $panels_data['grids'][$gi]['style']['collapse_order'] : ( !is_rtl() ? 'left-top' : 'right-top' );
```
Comment (braamgenis) [inline siteorigin-panels.php]: Same question here as previously about `is_rtl` still causing it to toggle if the option set by a user.

File: siteorigin-panels.php [hunk 4]
```diff
@@ -1365,35 +1375,6 @@ function siteorigin_panels_render_form($widget, $instance = array(), $raw = fals
 }
 
 /**
- * This takes existing Page Builder data and makes it RTL by reversing the content
- */
-function siteorigin_panels_make_rtl($panels_data){
```
Comment (braamgenis) [inline siteorigin-panels.php]: Ah, ok, I see you removed `siteorigin_panels_make_rtl`. so you can ignore the previous comment about the added filter.

File: siteorigin-panels.php (hunk 1 above)
Comment (gregpriday) [inline siteorigin-panels.php]: I think once a user has manually set the collapse order, we should obey that fairly strictly. It's only when they've selected _Default_ that we'd want to use `is_rtl()` to work out the best option.

File: siteorigin-panels.php (hunk 2 above)
Comment (gregpriday) [inline siteorigin-panels.php]: That was the original intention, but the way we're reversing the cells now, on the processed version of the data, is a lot more effective.

https://github.com/siteorigin/siteorigin-panels/pull/133/files#diff-72e0505cdd6cb4ce7175f772c9c68bafR1018

It does mean this new filter isn't all that important though. We could remove it.

File: siteorigin-panels.php (hunk 1 above)
Comment (braamgenis) [inline siteorigin-panels.php]: Ok, I think it's fair enough either way. Just wondered about the case where a user might want the opposite of the deault `is_rtl()` behaviour.

File: siteorigin-panels.php (hunk 2 above)
Comment (braamgenis) [inline siteorigin-panels.php]: Cool. Then I don't think it will hurt to leave it in, but up to you.

**Comment by braamgenis** [comment]
Cool, all good. Will leave to you if you want to make any changes before merging.

**Comment by braamgenis** [comment]
Seem to be some conflicts, so I'll leave to you to merge.

**Pull request merged**